app/services/course/assessment/question/programming/python/python_package_service.rb

```diff
@@ -151,22 +151,21 @@ def generate_zip_file(data_files_to_keep)
       index = 1
       [:public, :private, :evaluation].each do |test_type|
         tests[test_type].try(:each) do |test|
-          test_fn = "    def test_#{test_type}_#{index}(self):\n"\
-                    "        self.meta['expression'] = #{test[:expression].inspect}\n"\
-
           # String types should be displayed with quotes, other types will be converted to string
           # with the str method.
           expected = string?(test[:expected]) ? test[:expected].inspect : "str(#{test[:expected]})"
-          test_fn += "        self.meta['expected'] = #{expected}\n"
-
-          unless test[:hint].blank?
-            test_fn += "        self.meta['hint'] = #{test[:hint].inspect}\n"
-          end
-
-          test_fn += "        _out = #{test[:expression]}\n"\
-                     "        self.meta['output'] = _out\n"\
-                     "        self.assertEqual(_out, #{test[:expected]})\n"\
-                     "\n"
+          hint = test[:hint].blank? ? String(nil) : "self.meta['hint'] = #{test[:hint].inspect}"
+
+          test_fn = <<-TEST_FN
```

Comment:
Can use `PYTHON` here - some IDEs actually recognise that and will highlight it python style.